text/XXX-reduction.md

```diff
@@ -0,0 +1,227 @@
+    *   Full normalization by default
+    *   Does not support `Arguments`
+    *   Supports all reduction flags
+    *   Does not support fixpoint refolding ([\[Experiment\] Hack a global fixpoint expansion in conversion. by ppedrot · Pull Request #18672 · coq/coq](https://github.com/coq/coq/pull/18672) )
+    *   Largely coincides with the default cast except for reduction flags that
+        will not be recorded in the cast.
+*   `cbv`, `compute`
+    *   ???
+    *   Eager reduction
+    *   Does not support `Arguments`
+    *   Supports all reduction flags
+    *   Does not support fixpoint refolding
+    *   Default cast will coincide in terms of the result but will use kernel
+        reduction/conversion to get there.
+*   `hnf`
+    *   Head normal form
+    *   Does not support Arguments (❔)
+    *   Does not support reduction flags (❔)
+    *   Supports fixpoint refolding
+*   `simpl`
+    *   Rocq’s premier simplification engine
+    *   Does not unfold aliases unless they uncover reduction steps
+    *   Performance mixed
+        *   Less efficient than `lazy` across the board but usually better than `cbn`
+        *   Issues with primitive projections ([Primitive projections introduce slowdown in cbn and simpl · Issue #15720 · coq/coq](https://github.com/coq/coq/issues/15720) )
+        *   Issues with nested fixpoints ([simpl is exponential in the number of nested fixpoints · Issue #13772 · coq/coq](https://github.com/coq/coq/issues/13772) )
+        *   Issues without fixpoints or primitive projections ([!simpl scales exponentially even without nested fixpoints · Issue #16397 · coq/coq](https://github.com/coq/coq/issues/16397) )
+    *   Partial support for `Arguments`
+        *   `simpl` will ignore `Arguments : simpl never` in `match <HERE> with`
+    *   Only supports `head` reduction flag
+    *   Supports fixpoint refolding
+    *   No dedicated cast, relies on default cast and thus kernel
+        reduction/conversion at `Qed` time which could lead to performance
+        issues
+*   `cbn`
+    *   An alternative to `simpl`
+    *   Eagerly reduces aliases even when they uncover no reduction steps
+    *   Performance is bad
+        *   Slower than any other reduction mechanism often by orders of magnitude
+        *   Similar issues as with `simpl`, just worse
+    *   Full support for `Arguments`
+    *   Supports fixpoint refolding
+        *   Also refolds aliases (❔)
+    *   No dedicated cast, relies on default cast and thus kernel
+        reduction/conversion at `Qed` time which could lead to performance
+        issues
+
+
+# Problematic Reduction Examples
+
+## <a id="reif"></a> Reification with User Terms
+Iris Proof Mode and BlueRock’s automation use reflection and computation to
+perform changes to the shallowly embedded iris goal.
+
+### Requirements
+
+*   Must only reduce computation introduced by the automation tactics
+    *   Can be addressed with whitelist reduction flags, or
```

Comment:
Alone, this isn't really really enough. You really need an **and** here since functions that need to be reduced often use parts of the standard library such as list append and such.

```suggestion
    *   Can be addressed with whitelist reduction flags, **and**
```